src/java.base/share/classes/sun/security/ssl/X509Authentication.java

```diff
@@ -270,8 +270,8 @@ private static SSLPossession createClientPossession(
     private static SSLPossession createServerPossession(
             ServerHandshakeContext shc, String[] keyTypes) {
         X509ExtendedKeyManager km = shc.sslContext.getX509KeyManager();
-        String serverAlias = null;
         for (String keyType : keyTypes) {
+            String serverAlias = null;
```

Comment:
It seems not a simple cleanup. If you move the `serverAlias` definition into the loop, it means everytime entering into loop, the `serverAlias` will be `null`. But the previous code can keep the `serverAlias` value to the next iteration. So the meaning of the code has been changed. I don't know which meanings is right because I don't have a deep understanding of SSL now.

Reply:
With my understanding, in each iteration, firstly choose an alias from key manager with a key type, and then try to get the keys and certificates associated with this alias.
If an alias or its associated keys and certificates have something wrong, it should try other aliases associated with the key types in the remaining iterations.

If an alias can be used by the subsequent iterations, that looks a bug.

Reply:
> If an alias can be used by the subsequent iterations, that looks a bug.

Looks like a bug. So your patch is a bug fix instead of simple cleanup. Should we change the title of this issue or/and provide a test case?

Reply:
At the beginning of a iteration, `km.chooseServerAlias` or `km.chooseEngineServerAlias` tries to find an alias. Then, `serverAlias` should be `null` or an existing alias in the key manager.
The `serverAlias` assigned by the last iteration should not used by the current iteration.
So, I don't see a bug at the moment.

Reply:
I know what I missed now. The interface `SSLTransport` only has two implementations `SSLSocketImpl` and `SSLEngineImpl`, so the `serverAlias` is always set at the beginning of the loop. Thanks for your kindly explanation.